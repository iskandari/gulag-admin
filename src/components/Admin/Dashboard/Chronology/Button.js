import styled from 'styled-components';

export default styled.button`
  width: 100%;
  padding: 15px 0 17px;
  border: none;
  margin: 0;

  text-align: left;

  background-color: transparent;

  &:hover {
    background-color: #f3f3f3;
  }
`;
