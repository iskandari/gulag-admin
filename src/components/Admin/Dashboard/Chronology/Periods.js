import styled from 'styled-components';

export default styled.div`
  display: inline-block;
  padding: 15px 0 11px;

  column-count: 2;
  column-gap: 60px;
`;
