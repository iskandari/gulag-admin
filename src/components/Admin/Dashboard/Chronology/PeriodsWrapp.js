import styled from 'styled-components';

export default styled.div`
  width: 100%;
  &:hover {
    background-color: #f3f3f3;
  }
`;
