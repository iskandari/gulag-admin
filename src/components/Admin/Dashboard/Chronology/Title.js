import styled from 'styled-components';

export default styled.div`
  width: 100%;
  padding: 5px 0;
  border-bottom: 1px solid rgba(0, 0, 0, .3);

  font-weight: 900;
  font-size: 48px;
`;
