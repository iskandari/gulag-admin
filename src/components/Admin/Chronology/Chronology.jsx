import React from 'react';
import PropTypes from 'prop-types';

import Header from './Header/Header';
import Periods from './Periods/Periods';

// styled
import Container from './Container';

const Chronology = props => (
  <Container>
    <Header goBack={props.pushToDashboard} />
    <Periods
      periods={props.periods}
      createPeriod={props.createPeriod}
      deletePeriod={props.deletePeriod}
    />
  </Container>
);

Chronology.propTypes = {
  pushToDashboard: PropTypes.func.isRequired,
  periods: PropTypes.object.isRequired,
  createPeriod: PropTypes.func.isRequired,
  deletePeriod: PropTypes.func.isRequired
};

export default Chronology;
