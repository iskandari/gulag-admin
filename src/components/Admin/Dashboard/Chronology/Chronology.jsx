import React from 'react';
import PropTypes from 'prop-types';

// styled
import Container from './Container';
import Title from './Title';
import PeriodsWrapp from './PeriodsWrapp';
import Periods from './Periods';
import Period from './Period';
import Button from './Button';

const Chronology = ({ periods, openChronology }) => (
  <Container>
    <Title>Хронология Гулага</Title>
    <PeriodsWrapp onClick={openChronology}>
      {periods.size !== 0 && (
        <Periods>
          {periods.map(period => (
            <Period>
              <span>{period.get('year')}</span>
              <span>{period.getIn(['title', 'ru'])}</span>
            </Period>
          ))}
        </Periods>
      )}
    </PeriodsWrapp>
    {periods.size === 0 && <Button onClick={openChronology}>Добавить периоды</Button>}
  </Container>
);

Chronology.propTypes = {
  periods: PropTypes.object.isRequired,
  openChronology: PropTypes.func.isRequired
};

export default Chronology;
