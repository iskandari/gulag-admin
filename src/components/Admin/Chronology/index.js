import { createSelector } from 'reselect';
import { Map } from 'immutable';
import { push } from 'react-router-redux';
import { connect } from 'react-redux';

import { periodsSelector } from '../../App/dataSelectors';
import { createPeriod, deletePeriod } from '../../App/dataReducer';

import Chronology from './Chronology';

const mapStateToProps = () =>
  createSelector(periodsSelector, periods => ({
    periods: periods.reduce((acc, period) => acc.set(period.get('id'), period), Map())
  }));

const mapDispatchToProps = dispatch => ({
  pushToDashboard: () => dispatch(push('/admin')),
  createPeriod: () => dispatch(createPeriod()),
  deletePeriod: id => dispatch(deletePeriod(id))
});

export default connect(mapStateToProps, mapDispatchToProps)(Chronology);
