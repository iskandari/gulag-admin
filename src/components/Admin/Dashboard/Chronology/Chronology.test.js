import React from 'react';
import renderer from 'react-test-renderer';

import Chronology from './Chronology';

it('renders', () => {
  const renderedValue = renderer.create(<Chronology />)
    .toJSON();
  expect(renderedValue)
    .toMatchSnapshot();
});
