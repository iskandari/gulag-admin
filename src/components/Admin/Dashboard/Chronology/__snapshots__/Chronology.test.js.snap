// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="sc-bdVaJa eUGucI"
>
  <div
    className="sc-bwzfXH fTzidj"
  >
    Хронология Гулага
  </div>
</div>
`;
